src/main/java/school/lemon/changerequest/java/introduction/hw2/Task8.java

```diff
@@ -3,6 +3,30 @@
 public class Task8 {
 
     public static int[] evenOdd(int[] arr) {
-        return null;
+        if (arr==null){
+            return null;
+        }
+        int len = arr.length;
+        int chet[] = new int[len];
+        int nechet[] = new int[len];
+        int j = 0;
+        int k = 0;
+        for (int i = 0; i < len; i++) {
+                if (arr[i] % 2 == 0) {
+                    chet[j++] = arr[i];
+                } else {
+                    nechet[k++] = arr[i];
+                }
+            }
+                k = 0;
+                for (int i = 0; i < len ; i++) {
+                    if (i < j ) {
+                        arr[i] = chet[i];
+                    } else {
+                        arr[i] = nechet[k++];
+                    }
+                }
+                return arr;
+            }
```

Comment:
There is no need for 2 additional arrays. You can try to solve this task using 1 additional(result) array and 2 for-loops.